src/schema.ts

```diff
@@ -37,6 +39,8 @@ export class SubmitResult {
     | SubmitResultV2
     | SubmitResultV1
     | LegacyExecutionResult;
+  public gasBurned: GasBurned;
+  public tx: string | undefined;
```

Comment:
This is not great. As nothing has changed in terms of the engine ABI, there should be no reason to modify `schema.ts`. Right now, the suggested changes will lead to a lot of confusion for what is actually part of the ABI.